### DISCUSSION TITLE
npx prisma migrate dev --create-only : not working as intended in documentations

### DISCUSSION THREAD

**Discussion by smrarote** [prisma/prisma#22273] [Q&A] [answered]
### Bug description

Working with prisma 5.6.0 on dev environment intended behaviour is first create migration and then apply it on the dev environment.
npx prisma migrate dev --create-only command used from documentation [DOC](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#:~:text=If%20it%20detects%20changes%20to,(for%20example%2C%20Prisma%20Client))

![image](https://github.com/prisma/prisma/assets/59783361/5533e9a4-36a1-45ca-9ac5-a6f43a932257)

Need to crate the migration first confirm it, may be review and apply custom changes and them apply on the dev environment, further using the migration.sql for production and test db's 

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Ubuntu
- Database: PostgreSQL
- Node.js version: v20

### Prisma Version

```
5.6.0

```

**Comment by janpio** [comment, marked as answer]
Are you sure this is not caused by your usage of `dotenv` as the first command, which wraps `prisma` and "swallows" all the options with `--`?
 I think you need to do `-- --create-only` in this case, possible?

**Reply by smrarote** [reply]
@janpio Fixed the issue for me !

**Reply by janpio** [reply]
I am still wondering if there is a way to surface that to our users automatically - it is a _very_ simple mistake to make. But as far as I know, Prisma does not learn about `--create-only` existing in this case, at all :/
